src/sentences/utils.rs

```diff
@@ -49,6 +49,43 @@ pub(crate) fn parse_hms(i: &str) -> IResult<&str, NaiveTime> {
     )(i)
 }
 
+/// The number of milliseconds in a second.
+const MILLISECS_PER_SECOND: i64 = 1000;
+/// The number of milliseconds in a minute.
+const MILLISECS_PER_MINUTE: i64 = 60000;
+/// The number of milliseconds in a hour.
+const MILLISECS_PER_HOUR: i64 = 3600000;
+
+pub(crate) fn parse_duration_hms(i: &str) -> IResult<&str, Duration> {
+    map_res(
+        tuple((
+            map_res(take(2usize), parse_num::<i64>),
+            map_res(take(2usize), parse_num::<i64>),
+            map_parser(take_until(","), double),
+        )),
+        |(hour, minutes, sec)| -> core::result::Result<Duration, &'static str> {
+            if sec.is_sign_negative() {
+                return Err("Invalid time: second is negative");
+            }
+            if hour >= 24 {
+                return Err("Invalid time: hour >= 24");
+            }
+            if minutes >= 60 {
+                return Err("Invalid time: min >= 60");
+            }
+            if sec >= 60. {
+                return Err("Invalid time: sec >= 60");
+            }
+            Ok(Duration::milliseconds(
+                hour * MILLISECS_PER_HOUR
+                    + minutes * MILLISECS_PER_MINUTE
+                    + (sec.trunc() as i64) * MILLISECS_PER_SECOND
+                    + (sec.fract() * 1_000f64).round() as i64,
```

Comment:
- [ ] It's good to include checked operations here to avoid overflows.

Reply:
Yes, you're right actually. Since there are some checks regarding the values above this code maybe it's better to keep it simple.

I still prefer to have `u8` for the hours and we could even improve performance for non-64bit systems using `f32` for seconds.
Adding the rest of the checks for negative values will be unnecessary for `hours` and we still need to add `NaN` & negative (`-`) check for the `f32`.

Adding one small comment before creating the duration on why there's no chance of overflow should be enough. After all for hours we only take 2 chars which limits us to `99` as the maximum value.

What do you think?

Reply:
u8 and f32 updated. Comment added, so we should be OK here.